gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Acl.java

```diff
@@ -0,0 +1,400 @@
+  }
+
+  public static abstract class Entity implements Serializable {
+
+    private static final long serialVersionUID = 8111776788607959944L;
+
+    private final Type type;
+
+    public enum Type {
+      DOMAIN, GROUP, USER, VIEW
+    }
+
+    Entity(Type type) {
+      this.type = type;
+    }
+
+    public Type type() {
+      return type;
+    }
+
+    abstract Access toPb();
+
+    static Entity fromPb(Access access) {
+      if (access.getDomain() != null) {
+        return new Domain(access.getDomain());
+      }
+      if (access.getGroupByEmail() != null) {
+        return new Group(access.getGroupByEmail());
+      }
+      if (access.getSpecialGroup() != null) {
+        return new Group(access.getSpecialGroup());
+      }
+      if (access.getUserByEmail() != null) {
+        return new User(access.getUserByEmail());
+      }
+      if (access.getView() != null) {
+        return new View(TableId.fromPb(access.getView()));
+      }
+      // Unreachable
+      throw new BigQueryException(BigQueryException.UNKNOWN_CODE,
+          "Unrecognized access configuration", false);
+    }
+  }
+
+  /**
+   * Class for a BigQuery Domain entity.
+   */
+  public static final class Domain extends Entity {
+
+    private static final long serialVersionUID = -3033025857280447253L;
+    
+    private final String domain;
+
+    /**
+     * Creates a Domain entity given the domain name.
+     */
+    public Domain(String domain) {
+      super(Type.DOMAIN);
+      this.domain = domain;
+    }
+
+    /**
+     * Returns the domain name.
+     */
+    public String domain() {
+      return domain;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      Domain domainEntity = (Domain) o;
+      return Objects.equals(type(), domainEntity.type())
+          && Objects.equals(domain, domainEntity.domain());
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(type(), domain);
+    }
+
+    @Override
+    public String toString() {
+      return toPb().toString();
+    }
+
+    @Override
+    Access toPb() {
+      return new Access().setDomain(domain);
+    }
+  }
+
+  /**
+   * Class for a BigQuery Group entity.
+   */
+  public static final class Group extends Entity {
+
+    private static final String PROJECT_OWNERS = "projectOwners";
+    private static final String PROJECT_READERS = "projectReaders";
+    private static final String PROJECT_WRITERS = "projectWriters";
+    private static final String ALL_AUTHENTICATED_USERS = "allAuthenticatedUsers";
+    private static final long serialVersionUID = 5146829352398103029L;
+    
+    private final String identifier;
+
+    /**
+     * Creates a Group entity given its identifier. Identifier can be either a special group
+     * identifier ({@code projectOwners}, {@code projectReaders}, {@code projectWriters} and
```

Comment:
What you have now is great though I do wonder if you tried it?

Eclipse and intelliJ quick javadoc view seems to accept and use it but it is possible
that javadoc generation (which is defined for specific access level) will not work for it.
Just curious...